src/librustc/traits/project.rs

```diff
@@ -1247,6 +1247,34 @@ fn confirm_callable_candidate<'cx, 'gcx, 'tcx>(
                                               fn_sig,
                                               flag);
 
+    if selcx.projection_mode().is_any() {
```

Comment:
(for those following along this thread, niko left a comment below saying "I meant FulfillmentContext, but I may be incorrect.")